node/network/approval-distribution/src/lib.rs

```diff
@@ -428,11 +436,25 @@ impl State {
 			match entry.known_by.entry(peer_id.clone()) {
 				hash_map::Entry::Occupied(knowledge) => {
 					if knowledge.get().known_messages.contains(&fingerprint) {
+						tracing::debug!(
+							target: LOG_TARGET,
+							?source,
+							?peer_id,
+							?fingerprint,
+							"Duplicate assignment",
+						);
 						modify_reputation(ctx, peer_id, COST_DUPLICATE_MESSAGE).await;
 						return;
 					}
 				}
 				hash_map::Entry::Vacant(_) => {
+					tracing::debug!(
+						target: LOG_TARGET,
+						?source,
+						?peer_id,
+						?fingerprint,
+						"Assignment from unknown peer",
```

Comment:
the peer maybe known, but we don't know if it knows about the given block (yet or it finalized some higher block)